src/ecpool_worker.erl

```diff
@@ -128,12 +136,16 @@ init([Pool, Id, Mod, Opts]) ->
                    on_reconnect = ensure_callback(proplists:get_value(on_reconnect, Opts)),
                    on_disconnect = ensure_callback(proplists:get_value(on_disconnect, Opts))
                   },
+    {ok, State}.
+
+handle_call(init_connect, _From, State = #state{pool = Pool, id = Id}) ->
     case connect_internal(State) of
         {ok, NewState} ->
             gproc_pool:connect_worker(ecpool:name(Pool), {Pool, Id}),
-            {ok, NewState};
-        {error, Error} -> {stop, Error}
```

Comment:
maybe return ignore here, and send `error` message to collector

Reply:
Do you think this is better? In that case, why is that better?

Reply:
there is no need for the gen_call

Reply:
I will remove the call and do the initial connect in the init function so this is irrelevant.